packages/api/endpoints/granules.js

```diff
@@ -678,6 +706,108 @@ const associateExecution = async (req, res) => {
   });
 };
 
+const BulkPatchGranuleCollectionSchema = z.object({
+  apiGranules: z.array(z.object({}).catchall(z.any())).nonempty(),
+  collectionId: z.string().nonempty(),
+  esConcurrency: z.number().positive(),
+}).catchall(z.unknown());
+
+const BulkPatchSchema = z.object({
+  apiGranules: z.array(z.object({}).catchall(z.any())).nonempty(),
+  dbConcurrency: z.number().positive(),
+  dbMaxPool: z.number().positive(),
+}).catchall(z.unknown());
+
+const parseBulkPatchGranuleCollectionPayload = zodParser('BulkPatchGranuleCollection payload', BulkPatchGranuleCollectionSchema);
+const parseBulkPatchPayload = zodParser('BulkPatchSchema payload', BulkPatchSchema);
+
+/**
+ * Update a batch of granules to change collectionId to a new collectionId
+ * in PG and ES
+ *
+ * @param {Object} req - express request object
+ * @param {Object} req.testContext - test context for client requests
+ * @param {Object} req.body - request body for patching a granule
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function bulkPatchGranuleCollection(req, res) {
+  const {
+    collectionPgModel = new CollectionPgModel(),
+    knex = await getKnexClient(),
+    esClient = await getEsClient(),
+    mappingFunction = pMap,
+  } = req.testContext || {};
+  req.body.esConcurrency = req.body.esConcurrency ?? 5;
+  const body = parseBulkPatchGranuleCollectionPayload(req.body);
+  if (isError(body)) {
+    return returnCustomValidationErrors(res, body);
+  }
+
+  const granules = req.body.apiGranules;
+  const granuleIds = granules.map((granule) => granule.granuleId);
+  const newCollectionId = req.body.collectionId;
+  const collection = await collectionPgModel.get(
+    knex,
+    deconstructCollectionId(newCollectionId)
+  );
+
+  await mappingFunction(
+    granules,
+    async (apiGranule) => {
+      await updateEsGranule(esClient, apiGranule, { collectionId: newCollectionId }, process.env.ES_INDEX, 'granule');
+    },
+    { concurrency: body.esConcurrency }
+  );
+  await updateBatchGranulesCollection(knex, granuleIds, collection.cumulus_id);
+
+  return res.send({
+    message: `Successfully wrote granules with Granule Id: ${granuleIds} to Collection Id: ${newCollectionId}`,
```

Comment:
Nit/usability: 

Should this be consumable via the upstream caller?  Not critical to the existing feature, but this essentially looks like a log line, not a consumable payload. 

{ message: <>, [granuleIds] }?